src/code/z_play.c

```diff
@@ -1808,8 +1823,9 @@ void Play_TriggerRespawn(PlayState* this) {
 }
 
 s32 func_800C0CB8(PlayState* this) {
-    return (this->roomCtx.curRoom.meshHeader->base.type != MESH_HEADER_TYPE_1) && (YREG(15) != 0x20) &&
-           (YREG(15) != 0x30) && (YREG(15) != 0x40) && (this->sceneNum != SCENE_HAIRAL_NIWA);
+    return (this->roomCtx.curRoom.meshHeader->base.type != MESH_HEADER_TYPE_1) &&
+           (R_SCENE_CAM_TYPE != SCENE_CAM_TYPE_HOUSE) && (R_SCENE_CAM_TYPE != SCENE_CAM_TYPE_FIXED) &&
+           (R_SCENE_CAM_TYPE != SCENE_CAM_TYPE_PIVOT) && (this->sceneNum != SCENE_HAIRAL_NIWA);
 }
```

Comment:
`MESH_HEADER_TYPE_1` is prerendered backgrounds, `SCENE_HAIRAL_NIWA` is the hyrule castle garden with guards (the stealth section)

this function is like "is not a fixed camera scene" ? I remember naming it something like that doing crude z_player docs

Reply:
I recognized the `MESH_HEADER_TYPE_1` -> `MESH_HEADER_TYPE_PRERENDERER` but interesting that the scene in there is the hyrule castle garden. There's a special camera setting for that scene, `CAM_SET_PREREND_SIDE_SCROLL` so it's slightly different to the other settings, but still a prerend I think?

Reply:
No the guards section isn't prerendered, but the camera there behaves quite uniquely. I assume it's very hardcoded for that specific scene

Reply:
Ah good to know. Then maybe the camera setting needs an update then